plugin/parser/differ/mysql/table_test.go

```diff
@@ -44,3 +44,228 @@ func TestTable(t *testing.T) {
+		{
+			old:  `CREATE TABLE book(id INT) DEFAULT COLLATE = latin1_swedish_ci;`,
+			new:  `CREATE TABLE book(id INT) DEFAULT COLLATE = utf8mb4_general_ci;`,
+			want: "ALTER TABLE `book` DEFAULT COLLATE = UTF8MB4_GENERAL_CI;\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) DEFAULT COLLATE = latin1_swedish_ci;`,
+			new:  `CREATE TABLE book(id INT);`,
+			want: "ALTER TABLE `book` DEFAULT COLLATE = UTF8MB4_GENERAL_CI;\n",
+		},
+		// CHECKSUM
+		{
+			old:  `CREATE TABLE book(id INT) CHECKSUM = 1;`,
+			new:  `CREATE TABLE book(id INT) CHECKSUM = 0;`,
+			want: "ALTER TABLE `book` CHECKSUM = 0;\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) CHECKSUM = 1;`,
+			new:  `CREATE TABLE book(id INT);`,
+			want: "ALTER TABLE `book` CHECKSUM = 0;\n",
+		},
+		// COMMENT
+		{
+			old:  `CREATE TABLE book(id INT) COMMENT = 'old';`,
+			new:  `CREATE TABLE book(id INT) COMMENT = 'new';`,
+			want: "ALTER TABLE `book` COMMENT = 'new';\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) COMMENT = 'old';`,
+			new:  `CREATE TABLE book(id INT);`,
+			want: "ALTER TABLE `book` COMMENT = '';\n",
+		},
+		// TODO(zp): handle drop COMPRESSION
+		{
+			old:  `CREATE TABLE book(id INT) COMPRESSION = 'ZLIB';`,
+			new:  `CREATE TABLE book(id INT) COMPRESSION = 'LZ4';`,
+			want: "ALTER TABLE `book` COMPRESSION = 'LZ4';\n",
+		},
+		// CONNECTION
+		{
+			old:  `CREATE TABLE book(id INT) CONNECTION = 'old';`,
+			new:  `CREATE TABLE book(id INT) CONNECTION = 'new';`,
+			want: "ALTER TABLE `book` CONNECTION = 'new';\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) CONNECTION = 'old';`,
+			new:  `CREATE TABLE book(id INT);`,
+			want: "ALTER TABLE `book` CONNECTION = '';\n",
+		},
+		// TODO(zp): handle drop DATA DIRECTORY
+		{
+			old:  `CREATE TABLE book(id INT) DATA DIRECTORY = 'old';`,
+			new:  `CREATE TABLE book(id INT) DATA DIRECTORY = 'new';`,
+			want: "ALTER TABLE `book` DATA DIRECTORY = 'new';\n",
+		},
+		// TODO(zp): handle drop INDEX DIRECTORY
+		{
+			old:  `CREATE TABLE book(id INT) INDEX DIRECTORY = 'old';`,
+			new:  `CREATE TABLE book(id INT) INDEX DIRECTORY = 'new';`,
+			want: "ALTER TABLE `book` INDEX DIRECTORY = 'new';\n",
+		},
+		// DELAY_KEY_WRITE
+		{
+			old:  `CREATE TABLE book(id INT) DELAY_KEY_WRITE = 1;`,
+			new:  `CREATE TABLE book(id INT) DELAY_KEY_WRITE = 0;`,
+			want: "ALTER TABLE `book` DELAY_KEY_WRITE = 0;\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) DELAY_KEY_WRITE = 1;`,
+			new:  `CREATE TABLE book(id INT)`,
+			want: "ALTER TABLE `book` DELAY_KEY_WRITE = 0;\n",
+		},
+		// ENCRYPTION
+		{
+			old:  `CREATE TABLE book(id INT) ENCRYPTION = 'Y';`,
+			new:  `CREATE TABLE book(id INT) ENCRYPTION = 'N';`,
+			want: "ALTER TABLE `book` ENCRYPTION = 'N';\n",
+		},
+		{
+			old:  `CREATE TABLE book(id INT) ENCRYPTION = 'Y';`,
+			new:  `CREATE TABLE book(id INT);`,
+			want: "ALTER TABLE `book` ENCRYPTION = 'N';\n",
+		},
+		// INSERT_METHOD
+		// TODO(zp): enable this test if the upstream repo fix it.
```

Comment:
You can add the PR link in the upstream repo here.

Reply:
Done.